new/sdk/networkservicemesh/common/heal/client.go

```diff
@@ -0,0 +1,145 @@
+package heal
+
+import (
+	"context"
+	"runtime"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/new/sdk/networkservicemesh/core/next"
+	"github.com/networkservicemesh/networkservicemesh/new/sdk/tools/extended_context"
+	"github.com/networkservicemesh/networkservicemesh/new/sdk/tools/serialize"
+)
+
+type healClient struct {
+	requestors        map[string]func()
+	closers           map[string]func()
+	reported          map[string]*connection.Connection
+	onHeal            []networkservice.NetworkServiceClient
+	client            connection.MonitorConnectionClient
+	eventReceiver     connection.MonitorConnection_MonitorConnectionsClient
+	updateExecutor    serialize.Executor
+	recvEventExecutor serialize.Executor
+	cancelFunc        context.CancelFunc
+}
+
+func NewClient(client connection.MonitorConnectionClient, onHeal ...networkservice.NetworkServiceClient) networkservice.NetworkServiceClient {
```

Comment:
Why `onHeal` is an array? Each item in that array is `networkservice.NetworkServiceClient` which is potentially implies a chain. So `onHeal` in other words is an array of chains. But seems like we need only one chain to rerun the request, don't we?

Reply:
Good point.  I've fixed this to only take one onHeal.